app/code/community/Bolt/Boltpay/Helper/CatalogHelper.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+class Bolt_Boltpay_Helper_CatalogHelper extends Mage_Core_Helper_Abstract
+{
+    /**
+     * @return Mage_Core_Model_Abstract
+     */
+    public function getSession()
+    {
+        return Mage::getSingleton('catalog/session');
+    }
+
+    /**
+     * Get Last real order ID from the catalog session
+     * @return integer
+     */
+    public function getLastRealOrderId()
+    {
+        return Mage::getSingleton('catalog/session')->getLastRealOrderId();
+    }
+
+    /**
+     * @param string $scope
+     * @return string
+     */
+    private function getQuoteIdKey($scope = 'ppc')
+    {
+        return $scope.'_quote_id_' . Mage::app()->getStore()->getId();
+    }
+
+    /**
+     * Get Quote for Product page
+     * @return Mage_Sales_Model_Quote
+     */
+    public  function getQuote()
+    {
+        $hasOrder = false;
+        $orderQuoteId = false;
+        $ppcQuoteId = $this->getSession()->getData($this->getQuoteIdKey());
+        if ($ppcQuoteId) {
+            $order = Mage::getModel('sales/order')->getCollection()->addAttributeToFilter('entity_id', $this->getLastRealOrderId())->setPageSize(1)->getFirstItem();
+            if ($order instanceof Mage_Sales_Model_Order) {
+                $orderQuoteId = $order->getQuoteId();
+            }
+            /** @var Mage_Sales_Model_Quote $ppcQuote */
+            $ppcQuote = Mage::getModel('sales/quote')->loadByIdWithoutStore($ppcQuoteId);
+            if ($orderQuoteId && $ppcQuoteId == $orderQuoteId) {
+                $ppcQuote->setIsActive(false);
+                $ppcQuote->delete();
+                $hasOrder = true;
+            }
+        }
+        if ($hasOrder || empty($ppcQuote)) {
+            /** @var Mage_Sales_Model_Quote $ppcQuote */
+            $ppcQuote = Mage::getModel('sales/quote');
+            $ppcQuote->setStore(Mage::app()->getStore());
+            $ppcQuote->reserveOrderId();
+        }
+        return $ppcQuote;
+    }
+
+    /**
+     * @return Mage_Sales_Model_Quote
+     */
+    public function getQuoteWithCurrentProduct(Mage_Catalog_Model_Product $product, $request = null)
+    {
+        /** @var Mage_Sales_Model_Quote $ppcQuote */
+        $ppcQuote = $this->getQuote();
+        $ppcQuote->removeAllItems();
+        $ppcQuote->addProduct($product, $request);
+        $ppcQuote->setParentQuoteId($ppcQuote->getId());
+        $ppcQuote->getBillingAddress();
+        $ppcQuote->getShippingAddress()->setCollectShippingRates(true);
+        $ppcQuote->setParentQuoteId($ppcQuote->getId());
+        $ppcQuote->collectTotals()->save();
+        $this->getSession()->setData($this->getQuoteIdKey(), $ppcQuote->getId());
+        
+        return $ppcQuote;
+    }
+
+    /**
+     * Get request for product add to cart procedure
+     *
+     * @param   mixed $requestInfo
+     * @return  Varien_Object
+     */
+    public function getProductRequest($requestInfo)
```

Comment:
naming nit: to me it seems more `convert` than `get`

Reply:
@daisy1754 i used naming from the magento core:
app/code/core/Mage/Checkout/Model/Cart.php
line 222
BTW this method does not convert in all cases.